[functorch] reclassifying max_unpool failures

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #83619
* __->__ #83618
* #83614
* #83612
* #83178
* #83597
* #83530

They're expected because this variant of max_unpool is non-determnistic

Also removed a skip that seems to pass alright on my machine

zou3519 committed:
[functorch] reclassifying max_unpool failures

They're expected because this variant of max_unpool is non-determnistic

Also removed a skip that seems to pass alright on my machine

[ghstack-poisoned]

samdow commented on functorch/test/test_ops.py:
```diff
@@ -661,11 +660,13 @@ def test_vmapvjp(self, device, dtype, op):
         # and causes different runs to produce different results.
         # skip because this is flaky depending on what the max_norm is!
         skip('nn.functional.embedding', ''),
+        skip('nn.functional.max_unpool1d'),  # this opinfo variant is nondeterministic
+        skip('nn.functional.max_unpool2d'),  # this opinfo variant is nondeterministic
+        skip('nn.functional.max_unpool3d'),  # this opinfo variant is nondeterminstic
```
Yikes! Thanks for finding this

In case this is gets fixed, would it make sense for us to add a skip if the corresponding autograd test is in the skips? (i.e. we skip here because gradgrad is skipped...though I'm not sure if the fact that these pass test_vjp but don't pass test_fn_grad means anything with this argument)

zou3519 replied:
Hmm, you bring up a good point. The problem isn't exactly that the opinfo variant is nondeterministic; it looks like it's skipped in autograd tests because the finite differences that gradcheck doesn't can't handle it.

This needs some more investigation, I'll dig into it.

zou3519 replied:
https://github.com/pytorch/pytorch/issues/80827, the comment in the OpInfo is wrong (max_unpool2d double backward / jvp is non-deterministic and this is a bug). Since it's a bug I'm going to update the comment

> In case this is gets fixed, would it make sense for us to add a skip if the corresponding autograd test is in the skips?

It might be a bit difficult to add a skip if something else has a skip, so, I'll just write a one-off test for it

zou3519 committed:
Update on "[functorch] reclassifying max_unpool failures"


They're expected because this variant of max_unpool is non-determnistic

Also removed a skip that seems to pass alright on my machine

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] reclassifying max_unpool failures"


They're expected because this variant of max_unpool is non-determnistic

Also removed a skip that seems to pass alright on my machine

[ghstack-poisoned]
